python/moltres_xs.py

```diff
@@ -0,0 +1,269 @@
+                    self.xs_lib[i][j][k]["FISSXS"] = list(
+                        data['INF_FISS'][index][::2])
+                    self.xs_lib[i][j][k]["NSF"] = list(
+                        data['INF_NSF'][index][::2])
+                    self.xs_lib[i][j][k]["FISSE"] = list(
+                        data['INF_KAPPA'][index][::2])
+                    self.xs_lib[i][j][k]["DIFFCOEF"] = list(
+                        data['INF_DIFFCOEF'][index][::2])
+                    self.xs_lib[i][j][k]["RECIPVEL"] = list(
+                        data['INF_INVV'][index][::2])
+                    self.xs_lib[i][j][k]["CHI"] = list(
+                        data['INF_CHIT'][index][::2])
+                    self.xs_lib[i][j][k]["CHI_D"] = list(
+                        data['INF_CHID'][index][::2])
+                    self.xs_lib[i][j][k]["BETA_EFF"] = list(
+                        data['BETA_EFF'][index][::2])
+                    self.xs_lib[i][j][k]["DECAY_CONSTANT"] = list(
+                        data['LAMBDA'][index][::2])
+                    self.xs_lib[i][j][k]["GTRANSFXS"] = list(
+                        data['INF_SP0'][index][::2])
+
+
+def read_input(fin):
+    with io.open(fin) as f:
+        lines = f.readlines()
+    k = 0
+    while k < len(lines):
+        line = lines[k]
+        if '[TITLE]' in line:
+            f = open(lines[k+1].split()[0], 'w')
+            k += 1
+
+        if '[MAT]' in line:
+            mat_dict = {}
+            num_mats = int(lines[k+1].split()[0])
+            val = lines[k+2].split()
+            for i in range(num_mats):
+                mat_dict[val[i]] = {'temps': [],
+                                    'file': [],
+                                    'uni': [],
+                                    'burn': [],
+                                    'bran': []
+                                    }
+            k += 2
+        if '[BRANCH]' in line:
+            tot_branch = int(lines[k+1].split()[0])
+            for i in range(tot_branch):
+                val = lines[k+2+i].split()
+                mat_dict[val[0]]['temps'].extend(
+                    [int(val[1])])
+                mat_dict[val[0]]['file'].extend(
+                    [int(val[2])])
+                mat_dict[val[0]]['burn'].extend(
+                    [int(val[3])])
+                mat_dict[val[0]]['uni'].extend(
+                    [int(val[4])])
+                mat_dict[val[0]]['bran'].extend(
+                    [int(val[5])])
+            k += 1+tot_branch
+
+        if 'FILES' in line:
+            num_files = int(lines[k+1].split()[0])
+            files = {}
+            for i in range(num_files):
+                XS_in, XS_t = lines[k+2+i].split()
+                if int(XS_t) == 1:
+                    files[i] = scale_xs(XS_in)
+                elif int(XS_t) == 2:
+                    files[i] = serpent_xs(XS_in)
+                else:
+                    raise("XS data not understood\n 1=scale \n 2=serpent")
```

Comment:
Why not use the words "scale" and "serpent" rather than identifying them with numbers? Makes the input more readable then.